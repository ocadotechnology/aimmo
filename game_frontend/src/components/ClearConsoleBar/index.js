import { Button, Toolbar } from '@material-ui/core';
import ClearIcon from 'components/icons/Clear';
import React from 'react';
import styled from 'styled-components'
import PropTypes from 'prop-types'

export const StyledClearConsoleBar = styled(Toolbar)`
  background: #2f3129;
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
`

export const StyledConsoleTitle = styled.div`
  display: inline-flex;
`

export const ClearButton = styled(Button)`
  right: 0vw;
`

const ClearConsoleBar = (props) => {
  return (
    <StyledClearConsoleBar variant="dense">
      <StyledConsoleTitle>
        <ClearButton
          variant="outlined"
          onClick={props.clearConsoleClicked}
          startIcon={<ClearIcon />}
        >
          Clear console
        </ClearButton>
      </StyledConsoleTitle>
    </StyledClearConsoleBar>
  )
}

ClearConsoleBar.propTypes = {
  clearConsoleClicked: PropTypes.func,
}

export default ClearConsoleBar;
