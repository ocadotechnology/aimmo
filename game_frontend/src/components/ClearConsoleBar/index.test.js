/* eslint-env jest */
import React from 'react'
import ClearConsoleBar from 'components/ClearConsoleBar'
import createShallowWithTheme from 'testHelpers/createShallow'

describe('<ClearConsoleBar />', () => {
  it('renders correctly', () => {
    const tree = createShallowWithTheme(<ClearConsoleBar />)

    expect(tree).toMatchSnapshot()
  })
})
