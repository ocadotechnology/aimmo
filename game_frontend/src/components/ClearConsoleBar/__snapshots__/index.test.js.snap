// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClearConsoleBar /> renders correctly 1`] = `
<Styled(WithStyles(ForwardRef(Toolbar)))
  variant="dense"
>
  <styled.div>
    <Styled(WithStyles(ForwardRef(Button)))
      startIcon={<ClearIcon />}
      variant="outlined"
    >
      Clear console
    </Styled(WithStyles(ForwardRef(Button)))>
  </styled.div>
</Styled(WithStyles(ForwardRef(Toolbar)))>
`;
